stdpopsim/ext/genomic_elements.py

```diff
@@ -0,0 +1,22 @@
+import attr
+
+@attr.s(kw_only=True)
+class GenomicElementType(object):
+    """
+    TODO: docstring.
+
+    See SLiM documentation for initializeGenomicElementType().
+    """
+    mutation_types = attr.ib(default=[stdpopsim.ext.MutationType()], type=list)
```

Comment:
ah, thanks for this! I have never ussed `attrs`. 

I think we need a list of mutation types so we can check for consistency (e.g. sum of proportions must be at most 1). Also, from here on mutation types should always be tied to a genomic element type, that is `simulate` will take a list of genomic element types instead of a list of mutation types). But let's see how this plays out later.

Reply:
I agree with you re list of indexes now! We can do the consistency checking within the contig.